data/data/install.openshift.io_installconfigs.yaml

```
@@ -2773,6 +2773,10 @@ spec:
                 required:
                 - hosts
                 type: object
+              external:
+                description: External is the empty configuration used when an external
+                  platform provider is used.
+                type: object
```

Maybe we need to create the properties too?

```suggestion
              external:
                description: External is the configuration used when an external
                    platform provider is used.
                properties:
                  platformName:
                    description: defines the arbitrary string represented cloud
                      provider name.
                    type: string
                type: object
```

---

we might want to add the [CloudControllerManagerState](https://github.com/openshift/api/pull/1434/files#diff-07b264a49084976b670fb699badaca1795027d6ea732a99226a5388104f6174fR347) as well otherwise we won't expect any CCM by default.

---

> Maybe we need to create the properties too?

Platform none appears to use the same minimal platform spec.  Are there other properties we want to include?

---

> we might want to add the [CloudControllerManagerState](https://github.com/openshift/api/pull/1434/files#diff-07b264a49084976b670fb699badaca1795027d6ea732a99226a5388104f6174fR347) as well otherwise we won't expect any CCM by default.

@adriengentil , we would include this in the infrastructure status, right?

---

I don't know if it's worth to make it configurable, but yes, it should be in the infrastructure object :+1:

---

> > Maybe we need to create the properties too?
> 
> @mtulio Platform none appears to use the same minimal platform spec. Are there other properties we want to include?

I think the unique attribute we may need to capture in the config is the `platformName`.

I will defer to @adriengentil and @elmiko to double-check.

---

yep, that makes sense, i missed that aspect of it